tests/unit/amo/components/TestFeaturedAddonReview.js

```diff
@@ -0,0 +1,184 @@
+import NestedStatus from 'react-nested-status';
+import * as React from 'react';
+
+import { fetchReview, setReview } from 'amo/actions/reviews';
+import FeaturedAddonReview, {
+  FeaturedAddonReviewBase,
+} from 'amo/components/FeaturedAddonReview';
+import AddonReviewCard from 'amo/components/AddonReviewCard';
+import { createApiError } from 'core/api';
+import { ErrorHandler } from 'core/errorHandler';
+import { dispatchClientMetadata, fakeReview } from 'tests/unit/amo/helpers';
+import {
+  createStubErrorHandler,
+  fakeI18n,
+  createFakeLocation,
+  shallowUntilTarget,
+  createContextWithFakeRouter,
+  createFakeHistory,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const getProps = ({
+    location = createFakeLocation(),
+    params,
+    ...customProps
+  } = {}) => {
+    return {
+      errorHandler: createStubErrorHandler(),
+      i18n: fakeI18n(),
+      location,
+      store,
+      ...customProps,
+    };
+  };
+
+  const render = ({ ...customProps } = {}) => {
+    const props = getProps(customProps);
+
+    return shallowUntilTarget(
+      <FeaturedAddonReview {...props} />,
+      FeaturedAddonReviewBase,
+      {
+        shallowOptions: createContextWithFakeRouter({
+          history: createFakeHistory(),
+        }),
+      },
+    );
+  };
+
+  it('fetches a review on mount', () => {
+    const reviewId = 1;
+    const dispatch = sinon.stub(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    render({
+      errorHandler,
+      reviewId,
+    });
+
+    sinon.assert.calledWith(
+      dispatch,
+      fetchReview({
+        reviewId,
+        errorHandlerId: errorHandler.id,
+      }),
+    );
+  });
+
+  it('fetches a review when the reviewId changes', () => {
+    const firstReviewId = 1;
+    const secondReviewId = 2;
+    const dispatch = sinon.stub(store, 'dispatch');
```

Comment:
This test fails if you load the first review (this is the setup I was suggesting in previous comments):

```diff
diff --git a/tests/unit/amo/components/TestFeaturedAddonReview.js b/tests/unit/amo/components/TestFeaturedAddonReview.js
index 303cd6f22..0fb68a360 100644
--- a/tests/unit/amo/components/TestFeaturedAddonReview.js
+++ b/tests/unit/amo/components/TestFeaturedAddonReview.js
@@ -75,8 +75,11 @@ describe(__filename, () => {
   it('fetches a review when the reviewId changes', () => {
     const firstReviewId = 1;
     const secondReviewId = 2;
-    const dispatch = sinon.stub(store, 'dispatch');
     const errorHandler = createStubErrorHandler();
+    store.dispatch(
+      fetchReview({ errorHandlerId: errorHandler.id, reviewId: firstReviewId }),
+    );
+    const dispatch = sinon.stub(store, 'dispatch');
 
     const root = render({
       errorHandler,
```

However, I can't actually figure out a way to break the site because of this omission in `loadDataIfNeeded()`. I thought I could load one review then click another ID to load another but that works. I think it works because the router unmounts the component each time. 

Because of that, I wonder if you should just delete this test and not worry about it until it's a problem. What do you think? I'm surprised it works as is, though, because I've seen this case pose a problem in other components.

Reply:
This is odd because there isn't even any logic in `FeaturedAddonReview` to deal with this scenario, so no wonder the test doesn't pass! I'm not sure why I even wrote it. Having said that, we should deal with it in the component so I've updated it to do that. I also changed the test. I think there's a flaw in your logic above, because you call `fetchReview` but not `setReview` so that will leave `loadingReview` as `true` which would cause the test to fail because it won't dispatch `fetchReview` if `loadingReview` is true.

I've fixed all of this in both the component and the test.

Reply:
Oh, yeah, duh. I thought I was dispatching `setReview()`.